object/object.go

```diff
@@ -39,6 +40,10 @@ func GVK(o client.Object) schema.GroupVersionKind {
 	return lo.Must(apiutil.GVKForObject(o, scheme.Scheme))
 }
 
+func New[T any]() T {
```

Comment:
I like this better, abstract it away so I never have to think about this again !